src/replication.c

```diff
@@ -3072,15 +3072,15 @@ istgt_lu_mempool_stats(char **resp)
 		TAILQ_FOREACH(r, &spec->rq, r_next) {
 			j_replica = json_object_new_object();
 			json_object_object_add(j_replica, "replica",
-			    json_object_new_int64(r->zvol_guid));
+			    json_object_new_uint64(r->zvol_guid));
 			json_object_object_add(j_replica, "in-flight read",
-			    json_object_new_int64(
+			    json_object_new_uint64(
 			    r->replica_inflight_read_io_cnt));
 			json_object_object_add(j_replica, "in-flight write",
-			    json_object_new_int64(
+			    json_object_new_uint64(
 			    r->replica_inflight_write_io_cnt));
 			json_object_object_add(j_replica, "in-flight sync",
-			    json_object_new_int64(
+			    json_object_new_uint64(
```

Comment:
I recommend use of string as type for the above to avoid any kind of issue at other side while parsing. Please let them (consumers of the json) know to use `json.Number` as type of the respective fields. They can easily convert it into respective types using `fieldname.Int64()` etc.

Reply:
yes. They can use parse according to JSON type. For `uint64`, we have added an API which will give json object for `uint64` as string JSON object.